components/eamxx/src/share/field/field_layout.cpp

```diff
@@ -27,7 +27,10 @@ bool FieldLayout::is_tensor_layout () const {
   return lt==LayoutType::Tensor2D || lt==LayoutType::Tensor3D;
 }
 
-int FieldLayout::get_vector_dim () const {
+// get the index of the CMP (Components) tag in the FieldLayout
+// e.g., for FieldLayout f({COL, CMP, LEV}, {...});
+// we have get_component_idx(f) = 1
```

Comment:
Minor, but prob best to change `get_component_idx(f)` to `f.get_vector_component_idx()`?